crates/gitbutler-branch-actions/tests/dependencies.rs

```diff
@@ -731,6 +731,35 @@ fn dependecies_ignore_merge_commits() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn dependencies_handle_complex_branch_checkout() -> Result<()> {
+    let ctx = command_ctx("complex-branch-checkout")?;
+    let test_ctx = test_ctx(&ctx)?;
+    let default_target = test_ctx.virtual_branches.get_default_target()?;
+    let my_stack = &test_ctx.stack;
+
+    let dependencies = compute_workspace_dependencies(
+        &ctx,
+        &default_target.sha,
+        &HashMap::new(),
+        &test_ctx.all_stacks,
+    )?;
+
+    let commit_dependencies = dependencies.commit_dependencies.get(&my_stack.id).unwrap();
+    assert_commit_map_matches_by_message(
+        commit_dependencies,
+        HashMap::from([
+            ("update a again", vec!["update a"]),
+            ("update a", vec!["add a"]),
+            ("Merge branch 'delete-b' into my_stack", vec!["add b"]),
+        ]),
+        &ctx,
+        "Commit interdependencies correctly calculated. They should only pick up the merge commit when calculating dependencies",
```

Comment:
Should this say "incorrectly"?

Reply:
At least the intention is that the message states the correct behavior, so that if it fails, it shows what was supposed to happen.